src/node_snapshotable.cc

```diff
@@ -233,19 +233,19 @@ class FileReader : public FileIO {
 
     CHECK_GT(length, 0);  // There should be no empty strings.
     MallocedBuffer<char> buf(length + 1);
-    size_t r = fread(buf.data, 1, length + 1, f);
-    CHECK_EQ(r, length + 1);
+    memcpy(buf.data, sink->data() + read_total, length + 1);
     std::string result(buf.data, length);  // This creates a copy of buf.data.
 
     if (is_debug) {
-      Debug("\"%s\", read %d bytes\n", result.c_str(), r);
+      Debug("\"%s\", read %zu bytes\n", result.c_str(), length + 1);
     }
 
-    read_total += r;
+    read_total += length + 1;
     return result;
   }
 
   size_t read_total = 0;
+  const std::vector<char>* sink = nullptr;
```

Comment:
```suggestion
  const std::vector<char>* const sink = nullptr;
```
So it's not reassignable. Or make it a reference.

Reply:
Updated to a reference.